src/bindings/c/include/openvino/c/ov_partial_shape.h

```diff
@@ -27,48 +42,76 @@ typedef struct ov_partial_shape ov_partial_shape_t;
- * @param rank support dynamic and static rank
- * @param dims support dynamic and static dimension
+ * @param rank support dynamic and static rank.
+ * @param dims support dynamic and static dimension.
  *  Dynamic rank:
- *     Example: "?"
+ *     Example: `?`
  *  Static rank, but dynamic dimensions on some or all axes.
- *     Examples: "{1,2,?,4}" or "{?,?,?}" or "{1,2,-1,4}""
+ *     Examples: `{1,2,?,4}` or `{?,?,?}` or `{1,2,-1,4}`
  *  Static rank, and static dimensions on all axes.
- *     Examples: "{1,2,3,4}" or "{6}" or "{}""
+ *     Examples: `{1,2,3,4}` or `{6}` or `{}"`
  *
  * @param ov_status_e a status code.
  */
 OPENVINO_C_API(ov_status_e)
-ov_partial_shape_create(ov_partial_shape_t** partial_shape_obj, ov_rank_t* rank, ov_dimensions_t* dims);
+ov_partial_shape_init_dynamic_rank(ov_partial_shape_t* partial_shape_obj, ov_rank_t rank, ov_dimension_t* dims);
 
 /**
- * @brief Parse the partial shape to readable string.
+ * @brief Initialize a partial shape with static rank and static dimension.
  * @ingroup partial_shape
+ * @param rank support static rank.
+ * @param dims support static dimension.
+ *  Static rank, and static dimensions on all axes.
+ *     Examples: `{1,2,3,4}` or `{6}` or `{}`
+ *
  * @param ov_status_e a status code.
  */
-OPENVINO_C_API(const char*) ov_partial_shape_to_string(ov_partial_shape_t* partial_shape);
+OPENVINO_C_API(ov_status_e)
+ov_partial_shape_init_static(ov_partial_shape_t* partial_shape_obj, int64_t rank, int64_t* dims);
 
 /**
- * @brief Release partial shape.
+ * @brief Release internal memory allocated in partial shape.
  * @ingroup partial_shape
  * @param partial_shape will be released.
  */
-OPENVINO_C_API(void) ov_partial_shape_free(ov_partial_shape_t* partial_shape);
+OPENVINO_C_API(void) ov_partial_shape_deinit(ov_partial_shape_t* partial_shape);
 
 /**
- * @brief Covert partial shape to static shape.
+ * @brief Convert partial shape without dynamic data to a static shape.
  * @ingroup partial_shape
- * @param ov_status_e a status code, return OK if successful
+ * @param ov_status_e a status code, return OK if successful.
  */
 OPENVINO_C_API(ov_status_e) ov_partial_shape_to_shape(ov_partial_shape_t* partial_shape, ov_shape_t* shape);
 
 /**
- * @brief Covert shape to partial shape.
- * @ingroup shape
- * @param ov_status_e a status code, return OK if successful
+ * @brief Convert shape to partial shape.
+ * @ingroup partial_shape
+ * @param ov_status_e a status code, return OK if successful.
+ */
+OPENVINO_C_API(ov_status_e) ov_shape_to_partial_shape(ov_shape_t* shape, ov_partial_shape_t* partial_shape);
+
+/**
+ * @brief Helper function, convert a partial shape to readable string.
+ * @ingroup partial_shape
+ * @param ov_status_e a status code, return OK if successful.
  */
-OPENVINO_C_API(ov_status_e) ov_shape_to_partial_shape(ov_shape_t* shape, ov_partial_shape_t** partial_shape);
+OPENVINO_C_API(const char*) ov_partial_shape_to_string(const ov_partial_shape_t* partial_shape);
```

Comment:
I think is_dynamic() method can be useful for partial shape

Reply:
sure, I will add it.

Reply:
done